syne_tune/optimizer/schedulers/multiobjective/nsga2_searcher.py

```diff
@@ -0,0 +1,177 @@
+         metric is a list of strings specifying all objectives to be optimized.
+    :param points_to_evaluate: List of configurations to be evaluated
+        initially (in that order). Each config in the list can be partially
+        specified, or even be an empty dict. For each hyperparameter not
+        specified, the default value is determined using a midpoint heuristic.
+        If ``None`` (default), this is mapped to ``[dict()]``, a single default config
+        determined by the midpoint heuristic. If ``[]`` (empty list), no initial
+        configurations are specified.
+    :param mode: Should metric be minimized ("min", default) or maximized
+        ("max"). In the case of multi-objective optimization, mode can be a list defining for
+        each metric if it is minimized or maximized
+    :param population_size: Size of the population
+    """
+
+    def __init__(
+        self,
+        config_space: Dict[str, Any],
+        metric: List[str],
+        mode: Union[List[str], str],
+        points_to_evaluate: Optional[List[dict]] = None,
+        population_size: int = 20,
+        **kwargs,
+    ):
+        super(NSGA2Searcher, self).__init__(
+            config_space, metric, points_to_evaluate=points_to_evaluate, **kwargs
+        )
+        if isinstance(mode, str):
+            self._mode = [mode] * len(metric)
+        else:
+            self._mode = mode
+
+        self.hp_names = []
+        for hp_name, hp in config_space.items():
+            if isinstance(hp, Domain):
+                self.hp_names.append(hp_name)
+                assert type(hp) in [
+                    Categorical,
+                    Integer,
+                    Float,
+                    FiniteRange,
+                ], f"Type {type(hp)} for hyperparameter {hp_name} not supported."
+
+        self.problem = MultiObjectiveMixedVariableProblem(
+            config_space=config_space, n_var=len(self.hp_names), n_obj=len(metric)
+        )
+        self.algorithm = NSGA2(
+            pop_size=population_size,
+            sampling=MixedVariableSampling(),
+            mating=MixedVariableMating(
+                eliminate_duplicates=MixedVariableDuplicateElimination()
+            ),
+            eliminate_duplicates=MixedVariableDuplicateElimination(),
+        )
+        self.algorithm.setup(
+            problem=self.problem, termination=("n_eval", 2**32 - 1), verbose=False
+        )
+
+        self.current_population = self.algorithm.ask()
+        self.current_individual = 0
+        self.observed_values = []
+
+    def _update(self, trial_id: str, config: dict, result: dict):
+        observed_metrics = {}
+        for mode, metric in zip(self._mode, self._metric):
+            value = result[metric]
+            if mode == "max":
+                value *= -1
+            observed_metrics[metric] = value
+
+        self.observed_values.append(list(observed_metrics.values()))
```

Comment:
You sure you don't want to retain the dict?

Reply:
We have to pass a numpy array to pymoo